modules/react/select/stories/Select.stories.mdx

```diff
@@ -34,79 +34,209 @@ yarn add @workday/canvas-kit-react
 
 <ExampleCodeBlock code={Basic} />
 
+Our example uses [React state](<(https://react.dev/learn/state-a-components-memory)>) to track the
+value of the `Select`.
+
+### Label Position
+
+Set the `labelPosition` prop of the wrapping `FormField` to designate the position of the label
+relative to the `Select`. `labelPosition` accepts the following values:
+
+- `FormField.LabelPosition.Top` (Default)
+- `FormField.LabelPosition.Left`
+
+<ExampleCodeBlock code={LabelPosition} />
+
+### Required
+
+Set the `required` prop of the wrapping `FormField` to `true` to indicate that the field is
+required. Labels for required fields are suffixed by a red asterisk.
+
+<ExampleCodeBlock code={Required} />
+
 ### Disabled
 
-Set the `disabled` prop of the Select to prevent users from interacting with it.
+Set the `disabled` prop of `Select.Input` to prevent users from interacting with it.
 
 <ExampleCodeBlock code={Disabled} />
 
-You may also set the `disabled` prop of the Select Option to prevent users from selecting that
-Option. Note that the value of a Select may be programmatically set to a disabled Option by setting
-the Select's `value` prop to the value of the disabled Option. The disabled Option, however, will
-remain unselectable by the user.
+### Disabled Items
 
-<ExampleCodeBlock code={DisabledOption} />
+In order to disable items and prevent users from interacting with them:
 
-### Ref Forwarding
+1. Set the `nonInteractiveIds` prop of `Select` to an array of disabled item `id`s. This will
+   disable interaction for those items and exclude them from type-ahead.
 
-Select supports [ref forwarding](https://reactjs.org/docs/forwarding-refs.html). It will forward
-`ref` to its underlying `<select>` element.
+2. Set the `aria-disabled` attribute of all disabled `Select.Item`s to `true`. This ensures the
+   items are styled as disabled.
 
-<ExampleCodeBlock code={RefForwarding} />
+The following example assigns a `disabled` property to the disabled items in the `items` array.
+`disabled` is used to populate `nonInteractiveIds` and set `aria-disabled` for the disabled items.
+
+<ExampleCodeBlock code={DisabledOptions} />
+
+### With Icons
+
+Add `Select.Item.Icon` as a sub component of `Select.Item` if you want to render an icon for each
+item. The `Select.Item.Icon` takes in an `icon` prop of type `CanvasSystemIcon`. You can import
+icons from `@workday/canvas-system-icons-web`.
+
+If you want to render the selected item's icon in the `Select.Input` add the `inputStartIcon` prop.
+
+To get the current selected item's icon:
+
+- Register your `items` with the `useSelectModel`
+- Get current item by
+  `const currentItem = model.navigation.getItem(model.state.selectedIds[0], model)`
+- Pass the value to the input in the following way:
+  `<Select.Input inputStartIcon={currentItem.value.icon} id="form-field-input-id" />`
+
+**Notes: The `Select.Input` will only render an icon when there is an item selected.**
+
+<ExampleCodeBlock code={WithIcons} />
```

Comment:
A few updates for the `With Icons` example:

1. Cleaned up and updated language for consistency.
2. Linked to the `/assets/system-icons/` route. This content isn't great in SB, but it's better on the Canvas Site.
3. Moved the note about `Select.Input` to the end after the example and removed the bold since it didn't seem necessary to emphasize this.

```suggestion
### With Icons

Use `Select.Item.Icon` to render an icon for a `Select.Item`. The `icon` prop for `Select.Item.Icon` accepts
[system icons](/assets/system-icons/) from `@workday/canvas-system-icons-web`.

In order to render the icon for the selected item in the `Select.Input`:

1. Obtain a reference to the `model` by registering your `items` with `useSelectModel`.
4. Get the selected item: `const selectedItem = model.navigation.getItem(model.state.selectedIds[0], model)`
5. Pass the icon for the selected item to the input: `<Select.Input inputStartIcon={selectedItem.value.icon}>`

<ExampleCodeBlock code={WithIcons} />

Note that `Select.Input` will only render an icon if an item is selected.
```